Plugin/Model/OrderRepositoryPlugin.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Plugin\Model;
+
+use Magento\Sales\Model\OrderRepository;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\Data\OrderSearchResultInterface;
+use Ampersand\DisableStockReservation\Model\GetSourceSelectionResultFromOrder;
+use Magento\Sales\Api\Data\OrderExtensionFactory;
+
+/**
+ * Class OrderRepositoryPlugin
+ * @package Ampersand\DisableStockReservation\Plugin\Model
+ */
+class OrderRepositoryPlugin
+{
+    /**
+     * @var GetSourceSelectionResultFromOrder
+     */
+    private $getSourceSelectionResultFromOrder;
+
+    /**
+     * @var OrderExtensionFactory
+     */
+    private $orderExtensionFactory;
+
+    /**
+     * OrderRepositoryPlugin constructor.
+     * @param GetSourceSelectionResultFromOrder $getSourceSelectionResultFromOrder
+     * @param OrderExtensionFactory $orderExtensionFactory
+     */
+    public function __construct(
+        GetSourceSelectionResultFromOrder $getSourceSelectionResultFromOrder,
+        OrderExtensionFactory $orderExtensionFactory
+    ) {
+        $this->getSourceSelectionResultFromOrder = $getSourceSelectionResultFromOrder;
+        $this->orderExtensionFactory = $orderExtensionFactory;
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderInterface $result
+     * @return OrderInterface
+     */
+    public function afterGet(OrderRepository $subject, OrderInterface $result): OrderInterface
+    {
+        return $this->applyExtensionAttributesToOrder($result);
+    }
+
+    /**
+     * @param OrderRepository $subject
+     * @param OrderSearchResultInterface $result
+     * @return OrderSearchResultInterface
+     */
+    public function afterGetList(OrderRepository $subject, OrderSearchResultInterface $result): OrderSearchResultInterface
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrder($item);
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param OrderInterface $order
+     * @return OrderInterface
+     */
+    private function applyExtensionAttributesToOrder(OrderInterface $order): OrderInterface
+    {
+        $sourceSelectionResult = $this->getSourceSelectionResultFromOrder->execute($order);
```

Comment:
We should be loading from our repository/tables here, not calling the functionality to calculate the sources again. Also, be careful, we shouldn't be calling a DB query (which would be the case when using the repository methods) within a loop. That could potentially turn 1 query to get a list of orders into 1000's depending on how many orders are returned. We'd need to make only 1 additional query for all the order ID's returned from the `getList` method. Maybe look into collections and other repository list methods